lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -1270,6 +1272,118 @@ def get_workflow_menu(
+        return [WorkflowInvocationCollectionView(**invocation) for invocation in invocations]
+
+    @router.get(
+        "/api/invocations",
+        summary="Get the list of a user's workflow invocations.",
+        name="index_invocations",
+    )
+    def index_invocations(
+        self,
+        response: Response,
+        workflow_id: Annotated[Optional[DecodedDatabaseIdField], WorkflowIdQueryParam] = None,
+        history_id: Annotated[Optional[DecodedDatabaseIdField], HistoryIdQueryParam] = None,
+        job_id: Annotated[Optional[DecodedDatabaseIdField], JobIdQueryParam] = None,
+        user_id: Annotated[Optional[DecodedDatabaseIdField], UserIdQueryParam] = None,
+        sort_by: Annotated[Optional[InvocationSortByEnum], InvocationsSortByQueryParam] = None,
+        sort_desc: Annotated[Optional[bool], InvocationsSortDescQueryParam] = False,
+        include_terminal: Annotated[Optional[bool], InvocationsIncludeTerminalQueryParam] = True,
+        limit: Annotated[Optional[int], InvocationsLimitQueryParam] = None,
+        offset: Annotated[Optional[int], InvocationsOffsetQueryParam] = None,
+        instance: Annotated[Optional[bool], InvocationsInstanceQueryParam] = False,
+        view: Annotated[Optional[InvocationSerializationView], SerializationViewQueryParam] = None,
+        step_details: Annotated[Optional[bool], StepDetailQueryParam] = False,
+        trans: ProvidesUserContext = DependsOnTrans,
+    ) -> List[WorkflowInvocationCollectionView]:
+        """If workflow_id is supplied (either via URL or query parameter) it should be an
+        encoded StoredWorkflow id and returned invocations will be restricted to that
+        workflow. history_id (an encoded History id) can be used to further restrict the
+        query. If neither a workflow_id or history_id is supplied, all the current user's
+        workflow invocations will be indexed (as determined by the invocation being
+        executed on one of the user's histories)
+        :raises: exceptions.MessageException, exceptions.ObjectNotFound
+        """
+        invocation_payload = InvocationIndexPayload(
+            sort_by=sort_by,
+            sort_desc=sort_desc,
+            include_terminal=include_terminal,
+            limit=limit,
+            offset=offset,
+            instance=instance,
+        )
+        invocation_payload.workflow_id = workflow_id
+        invocation_payload.history_id = history_id
+        invocation_payload.job_id = job_id
+        invocation_payload.user_id = user_id
+        serialization_params = InvocationSerializationParams(
+            view=view,
+            step_details=step_details,
+        )
+        invocations, total_matches = self.invocations_service.index(trans, invocation_payload, serialization_params)
+        response.headers["total_matches"] = str(total_matches)
+        return [WorkflowInvocationCollectionView(**invocation) for invocation in invocations]
```

Comment:
```suggestion
        return invocations
```
These should already be WorkflowInvocationCollectionView